app/assets/javascripts/components/visual-editor.js

```diff
@@ -5,24 +5,63 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}
 ;(function (Modules) {
   function VisualEditor(module) {
     this.module = module
+  }
+
+  VisualEditor.prototype.init = function () {
+    this.content = this.module.querySelector('.app-c-visual-editor__content')
+    this.container = this.module.querySelector(
+      '.app-c-visual-editor__container'
+    )
+    this.textarea = this.module.querySelector(
+      '.app-c-visual-editor__govspeak-editor-wrapper textarea'
+    )
 
-    this.content = module.querySelector('.app-c-visual-editor__content')
-    this.container = module.querySelector('.app-c-visual-editor__container')
-    this.textarea = module.querySelector(
-      '.app-c-visual-editor__textarea-wrapper textarea'
+    new window.GovspeakVisualEditor(this.content, this.container, this.textarea) // eslint-disable-line no-new
+
+    this.govspeakEditorwrapper = this.module.querySelector(
+      '.app-c-visual-editor__govspeak-editor-wrapper'
+    )
+    this.visualEditorWrapper = this.module.querySelector(
+      '.app-c-visual-editor__visual-editor-wrapper'
+    )
+    this.exitButton = this.module.querySelector(
+      '.js-app-c-visual-editor__exit-button'
+    )
+    this.visual_editor_flag = document.querySelector(
+      '.app-c-visual-editor__hidden-field'
     )
+    this.contentEditable = this.container.querySelector(
+      'div[contenteditable="true"]'
+    )
+
+    this.exitButton.addEventListener('click', this.hideVisualEditor.bind(this))
+    this.showVisualEditor()
   }
 
-  VisualEditor.prototype.init = function () {
-    this.textarea.classList.add('app-c-visual-editor__textarea--hidden')
-    const id = this.textarea.getAttribute('id')
+  VisualEditor.prototype.showVisualEditor = function () {
+    this.govspeakEditorwrapper.classList.add(
+      'app-c-visual-editor__govspeak-editor-wrapper--hidden'
+    )
+    this.visualEditorWrapper.classList.add(
+      'app-c-visual-editor__visual-editor-wrapper--show'
+    )
+    this.visual_editor_flag.value = true
+
+    this.contentEditable.setAttribute('id', this.textarea.getAttribute('id'))
```

Comment:
The error summary link needs to point to this + we need to be able to give focus to the correct invalid element, and I believe it needs to be the body/textarea ID because the error pertains to a specific model field.